test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -56,17 +53,23 @@
 @SuppressForbidden(reason = "Uses a HttpServer to emulate an Azure endpoint")
 public class AzureHttpHandler implements HttpHandler {
     private static final Logger logger = LogManager.getLogger(AzureHttpHandler.class);
+    private static final Pattern RANGE_HEADER_PATTERN = Pattern.compile("^bytes=([0-9]+)-([0-9]+)$");
```

Comment:
We do this in at least two other places - I suggest making it a test util.

Reply:
Yep sounds good, I'll submit that in a separate PR though.

Reply:
WIP @ https://github.com/elastic/elasticsearch/pull/117304

Will wait till this is merged so we can put it on top